backend/src/services/chat-message/chat-message.service.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  ChatGetAllMessagesRequestDto,
+  ChatMessageCreateRequestDto,
+  ChatMessageGetAllItemResponseDto,
+  ChatMessageGetAllLastResponseDto,
+  ChatMessageGetAllResponseDto,
+  IdContainer,
+} from '~/common/types/types';
+import {
+  chatMessage as chatMessageRep,
+  menteesToMentors as menteesToMentorsRep,
+} from '~/data/repositories/repositories';
+import { createUuid, sanitizeHTML } from '~/helpers/helpers';
+
+type Constructor = {
+  chatMessageRepository: typeof chatMessageRep;
+  menteesToMentorsRepository: typeof menteesToMentorsRep;
+};
+
+class ChatMessage {
+  #chatMessageRepository: typeof chatMessageRep;
+
+  #menteesToMentorsRepository: typeof menteesToMentorsRep;
+
+  public constructor({
+    chatMessageRepository,
+    menteesToMentorsRepository,
+  }: Constructor) {
+    this.#chatMessageRepository = chatMessageRepository;
+    this.#menteesToMentorsRepository = menteesToMentorsRepository;
+  }
+
+  public async getAll({
+    chatId,
+  }: ChatGetAllMessagesRequestDto): Promise<ChatMessageGetAllResponseDto> {
+    const chatMessages = await this.#chatMessageRepository.getAll({
+      chatId,
+    });
+
+    return { items: chatMessages, chatId };
+  }
+
+  public async getAllLastMessages(
+    userId: number,
+  ): Promise<ChatMessageGetAllLastResponseDto> {
+    const usersMentorsDto: IdContainer[] =
+      await this.#menteesToMentorsRepository.getMentors(userId);
+    const usersMenteesDto: IdContainer[] =
+      await this.#menteesToMentorsRepository.getMentees(userId);
+
+    const chatOpponentsIds = [...usersMentorsDto, ...usersMenteesDto];
+
+    const lastMessagesWithMentorsAndMentees = await Promise.all(
+      chatOpponentsIds.map((chatOpponentIdConteiner) => {
+        return this.#chatMessageRepository.getLastMessage({
+          userId,
+          chatOpponentId: chatOpponentIdConteiner.id,
+        });
+      }),
+    );
```

Comment:
Wouldn't it be easier to work with `chatId` here? Just add `getAllChats` method, which is definitely needed (not sure if it's a part of your task), then get messages by only using `chatId`

Reply:
Again, was not my scope :)